288 tests fail

### Description

Here on NixOS, trying to update to version 5.3.3, we notice a lot of test failures.

At first, I observed these failures:

```
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0 -- /nix/store/lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/bin/python3.10
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/nix/store/yrdlpbs3hjc4d55yjphhr4w98bnx63y0-python3.10-astropy-5.3.3/.hypothesis/examples')

Running tests with Astropy version 5.3.3.
Running tests in lib/python3.10/site-packages/astropy.

Date: 2023-09-14T13:55:38

Platform: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.37

Executable: /nix/store/lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/bin/python3.10

Full Python Version: 
3.10.12 (main, Jun  6 2023, 22:43:10) [GCC 12.3.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.25.1
Scipy: not available
Matplotlib: not available
h5py: not available
Pandas: not available
PyERFA: 2.0.0.1
Cython: 0.29.36
Scikit-image: not available
asdf: not available

Using Astropy options: remote_data: none.

rootdir: /nix/store/yrdlpbs3hjc4d55yjphhr4w98bnx63y0-python3.10-astropy-5.3.3
plugins: xdist-3.3.1, astropy-0.10.0, hypothesis-6.68.2, arraydiff-0.5.0, astropy-header-0.2.2, cov-4.1.0, doctestplus-0.12.1, filter-subpackage-0.1.2, mock-3.11.1, openfiles-0.5.0, remotedata-0.4.0
initialized: 11/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
initialized: 12/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
initialized: 13/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
ready: 3/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
ready: 4/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
ready: 6/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
ready: 8/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
ready: 10/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
ready: 12/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
ready: 13/16 workers<frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 96 from PyObject
16 workers [0 items]    rrors                                                  
scheduling tests via LoadScheduling

==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
________________________ ERROR collecting test session _________________________
lib/python3.10/site-packages/astropy/coordinates/earth.py:214: in __new__
    self = cls.from_geocentric(*args, **kwargs)
E   TypeError: EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'

During handling of the above exception, another exception occurred:
lib/python3.10/site-packages/astropy/coordinates/earth.py:217: in __new__
    self = cls.from_geodetic(*args, **kwargs)
lib/python3.10/site-packages/astropy/coordinates/earth.py:317: in from_geodetic
    xyz = geodetic.to_cartesian().get_xyz(xyz_axis=-1) << height.unit
lib/python3.10/site-packages/astropy/coordinates/earth.py:956: in to_cartesian
    xyz = erfa.gd2gc(
../fxhxhikcjydm7sz2a3r2bs01sh8smbv9-python3.10-pyerfa-2.0.0.1/lib/python3.10/site-packages/erfa/core.py:16026: in gd2gc
    xyz, c_retval = ufunc.gd2gc(n, elong, phi, height)
E   TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'

During handling of the above exception, another exception occurred:
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
../lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
lib/python3.10/site-packages/astropy/table/tests/conftest.py:180: in <module>
    MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
lib/python3.10/site-packages/astropy/coordinates/earth.py:219: in __new__
    raise TypeError(
E   TypeError: Coordinates could not be parsed as either geocentric or geodetic, with respective exceptions "EarthLocation.from_geocentric() got an unexpected keyword argument 'lon'" and "operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'gd2gc'>, '__call__', 1, <Longitude [0., 1., 5., 6.] deg>, <Latitude [ 5.,  6., 10., 11.] deg>, <Quantity [0., 1., 2., 3.] m>): 'int', 'Longitude', 'Latitude', 'Quantity'"
=============================== warnings summary ===============================
../yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:1204
  /nix/store/yl17g92a32lpb013vaigx5q5js94mdn5-python3.10-pytest-7.4.0/lib/python3.10/site-packages/_pytest/config/__init__.py:1204: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: xdist
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
ERROR  - TypeError: Coordinates could not be parsed as either geocentric or geodetic...
======================== 1 warning, 16 errors in 4.21s =========================
```

<details><summary>Which led me applying this patch</summary>

```patch
diff --git i/astropy/table/tests/conftest.py w/astropy/table/tests/conftest.py
index 6d05a267a..d3469bf07 100644
--- i/astropy/table/tests/conftest.py
+++ w/astropy/table/tests/conftest.py
@@ -147,62 +147,6 @@ def table_type(request):
     return MaskedTable if request.param else table.Table
 
 
-# Stuff for testing mixin columns
-
-MIXIN_COLS = {
-    "quantity": [0, 1, 2, 3] * u.m,
-    "longitude": coordinates.Longitude(
-        [0.0, 1.0, 5.0, 6.0] * u.deg, wrap_angle=180.0 * u.deg
-    ),
-    "latitude": coordinates.Latitude([5.0, 6.0, 10.0, 11.0] * u.deg),
-    "time": time.Time([2000, 2001, 2002, 2003], format="jyear"),
-    "timedelta": time.TimeDelta([1, 2, 3, 4], format="jd"),
-    "skycoord": coordinates.SkyCoord(ra=[0, 1, 2, 3] * u.deg, dec=[0, 1, 2, 3] * u.deg),
-    "sphericalrep": coordinates.SphericalRepresentation(
-        [0, 1, 2, 3] * u.deg, [0, 1, 2, 3] * u.deg, 1 * u.kpc
-    ),
-    "cartesianrep": coordinates.CartesianRepresentation(
-        [0, 1, 2, 3] * u.pc, [4, 5, 6, 7] * u.pc, [9, 8, 8, 6] * u.pc
-    ),
-    "sphericaldiff": coordinates.SphericalCosLatDifferential(
-        [0, 1, 2, 3] * u.mas / u.yr, [0, 1, 2, 3] * u.mas / u.yr, 10 * u.km / u.s
-    ),
-    "arraywrap": ArrayWrapper([0, 1, 2, 3]),
-    "arrayswap": ArrayWrapper(np.arange(4, dtype="i").byteswap().newbyteorder()),
-    "ndarraylil": np.array(
-        [(7, "a"), (8, "b"), (9, "c"), (9, "c")], dtype="<i4,|S1"
-    ).view(table.NdarrayMixin),
-    "ndarraybig": np.array(
-        [(7, "a"), (8, "b"), (9, "c"), (9, "c")], dtype=">i4,|S1"
-    ).view(table.NdarrayMixin),
-    "stokescoord": coordinates.StokesCoord(range(1, 5)),
-}
-MIXIN_COLS["earthlocation"] = coordinates.EarthLocation(
-    lon=MIXIN_COLS["longitude"],
-    lat=MIXIN_COLS["latitude"],
-    height=MIXIN_COLS["quantity"],
-)
-MIXIN_COLS["sphericalrepdiff"] = coordinates.SphericalRepresentation(
-    MIXIN_COLS["sphericalrep"], differentials=MIXIN_COLS["sphericaldiff"]
-)
-
-
-@pytest.fixture(params=sorted(MIXIN_COLS))
-def mixin_cols(request):
-    """
-    Fixture to return a set of columns for mixin testing which includes
-    an index column 'i', two string cols 'a', 'b' (for joins etc), and
-    one of the available mixin column types.
-    """
-    cols = OrderedDict()
-    mixin_cols = deepcopy(MIXIN_COLS)
-    cols["i"] = table.Column([0, 1, 2, 3], name="i")
-    cols["a"] = table.Column(["a", "b", "b", "c"], name="a")
-    cols["b"] = table.Column(["b", "c", "a", "d"], name="b")
-    cols["m"] = mixin_cols[request.param]
-
-    return cols
-
 
 def _get_test_table():
     T = QTable.read(
```

</details>

And then I encountered the 288 tests failures. Here is the full log of it:

https://gist.github.com/doronbehar/fe22e0dd825c2fbb9285043b36119a4e

### Expected behavior

No tests fail.

### How to Reproduce

I tried build it on NixOS, using:

```
nix-build -A python3.pkgs.astropy https://github.com/doronbehar/nixpkgs/archive/35d2f581d34f9ea46b073a301cb8e98b56b98233.tar.gz
```

The build expression is available here:

https://github.com/doronbehar/nixpkgs/blob/35d2f581d34f9ea46b073a301cb8e98b56b98233/pkgs/development/python-modules/astropy/default.nix

### Versions

Versions are available in the log in gist, and also here: https://search.nixos.org/packages .

With pyerfa compiled with `-O2` (as was investigated in https://github.com/liberfa/pyerfa/issues/112 ), and with the patch I mentioned in my top most comment removed, only 1 test fails! And that is:

```

=================================== FAILURES ===================================
________ TestMaskedQuantityInitialization.test_masked_quantity_getting _________
[gw12] linux -- Python 3.10.12 /nix/store/lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/bin/python3.10

self = <astropy.utils.masked.tests.test_masked.TestMaskedQuantityInitialization object at 0x7fffb9afab00>

    def test_masked_quantity_getting(self):
>       mcls = Masked._masked_classes[self.a.__class__]
E       KeyError: <class 'astropy.units.quantity.Quantity'>

lib/python3.10/site-packages/astropy/utils/masked/tests/test_masked.py:305: KeyError
=========================== short test summary info ============================
FAILED lib/python3.10/site-packages/astropy/utils/masked/tests/test_masked.py::TestMaskedQuantityInitialization::test_masked_quantity_getting - KeyError: <class 'astropy.units.quantity.Quantity'>
= 1 failed, 22737 passed, 3611 skipped, 176 xfailed, 80 warnings in 107.95s (0:01:47) =
```

> For me, the test passes again if Iadd the following to the `TestMaskedQuantityInitialization` class

@mhvk Did you mean to apply this patch?

```diff
diff --git i/astropy/utils/masked/tests/test_masked.py w/astropy/utils/masked/tests/test_masked.py
index 1ca29088e..013cb33e8 100644
--- i/astropy/utils/masked/tests/test_masked.py
+++ w/astropy/utils/masked/tests/test_masked.py
@@ -290,6 +290,12 @@ class TestMaskedNDArraySubclassCreation:
 
 
 class TestMaskedQuantityInitialization(TestMaskedArrayInitialization, QuantitySetup):
+    @classmethod
+    def setup_class(self):
+        super().setup_class()
+        # Ensure we have used MaskedQuantity before - just in case a single test gets
+        # called; see gh-15316.
+        Masked(Quantity)
     def test_masked_quantity_class_init(self):
         # TODO: class definitions should be more easily accessible.
         mcls = Masked._masked_classes[self.a.__class__]
```

> Do you run the tests in a special way?

Up until now, I ran them with `astropy.test`.

> Re: `astropy.test()` and `sys.exit`
>
> [...]
>
> Or is this closer to what you are trying to do? [docs.pytest.org/en/stable/how-to/failures.html#stopping-after-the-first-or-n-failures](https://docs.pytest.org/en/stable/how-to/failures.html#stopping-after-the-first-or-n-failures)

@pllim so yes, using `pytest` directly, with or without `-x` should exit non cleanly when tests fail, I suppose. I tried that now, and all tests passed :tada:. However, our build system by default attempts to run `python -m pytest` instead of directly `pytest`, and the former causes the following error:

```
ImportError while loading conftest '/build/astropy-5.3.3/conftest.py'.
conftest.py:11: in <module>
    from astropy import __version__
astropy/__init__.py:174: in <module>
    log = _init_log()
astropy/logger.py:113: in _init_log
    log._set_defaults()
astropy/logger.py:497: in _set_defaults
    self.disable_warnings_logging()
astropy/logger.py:269: in disable_warnings_logging
    raise LoggingError(
E   astropy.logger.LoggingError: Cannot disable warnings logging: warnings.showwarning was not set by this logger, or has been overridden
```

We are discussing another issue related to `pytest` vs `python -m pytest` over at Nix' packages repo at https://github.com/NixOS/nixpkgs/issues/255262 . It would have been nice if running the test via `python -m pytest` was also supported.

--- a/astropy/utils/masked/tests/test_masked.py
+++ b/astropy/utils/masked/tests/test_masked.py
@@ -290,6 +290,12 @@ class TestMaskedNDArraySubclassCreation:
 
 
 class TestMaskedQuantityInitialization(TestMaskedArrayInitialization, QuantitySetup):
+    @classmethod
+    def setup_class(self):
+        super().setup_class()
+        # Ensure we have used MaskedQuantity before - just in case a single test gets
+        # called; see gh-15316.
+        Masked(Quantity)
     def test_masked_quantity_class_init(self):
         # TODO: class definitions should be more easily accessible.
         mcls = Masked._masked_classes[self.a.__class__]

--- a/astropy/utils/masked/tests/test_masked.py
+++ b/astropy/utils/masked/tests/test_masked.py
@@ -296,6 +296,7 @@ class TestMaskedQuantityInitialization(TestMaskedArrayInitialization, QuantitySe
         # Ensure we have used MaskedQuantity before - just in case a single test gets
         # called; see gh-15316.
         Masked(Quantity)
+
     def test_masked_quantity_class_init(self):
         # TODO: class definitions should be more easily accessible.
         mcls = Masked._masked_classes[self.a.__class__]

